LoopKit/CorrectionRangeOverrides.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  CorrectionRangeOverrides.swift
+//  LoopKit
+//
+//  Created by Rick Pasetto on 7/14/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import HealthKit
+import Foundation
+
+public struct CorrectionRangeOverrides: Equatable {
+    public enum Preset: Hashable, CaseIterable {
+        case preMeal
+        case workout
+    }
+
+    public var ranges: [Preset: ClosedRange<HKQuantity>]
+
+    public init(preMeal: DoubleRange?, workout: DoubleRange?, unit: HKUnit) {
+        ranges = [:]
+        ranges[.preMeal] = preMeal?.quantityRange(for: unit)
+        ranges[.workout] = workout?.quantityRange(for: unit)
+    }
+
+    public var preMeal: ClosedRange<HKQuantity>? { ranges[.preMeal] }
+    public var workout: ClosedRange<HKQuantity>? { ranges[.workout] }
+}
+
+extension CorrectionRangeOverrides.Preset {
```

Comment:
Should the titles of these presets be extracted, too? i.e. the strings 'Pre-Meal' and 'Workout'

Reply:
That would make sense, especially because from discussions with product, it seems like the "Pre-Meal" title may be changing in the future

Reply:
Yep, an oversight on my part.  Will do.